cmake/kokkos_functions.cmake

```diff
@@ -871,3 +871,46 @@ FUNCTION(KOKKOS_CHECK_DEPRECATED_OPTIONS)
     ENDIF()
   ENDFOREACH()
 ENDFUNCTION()
+
+# this function is provided to easily select which files use nvcc_wrapper:
+#
+#       GLOBAL      --> all files
+#       TARGET      --> all files in a target
+#       SOURCE      --> specific source files
+#       DIRECTORY   --> all files in directory
+#       PROJECT     --> all files/targets in a project/subproject
+#
+FUNCTION(kokkos_compilation)
```

Comment:
We've discussed this before, but didn't bother with it. I think we're reaching a critical code duplication point with this function appearing in two places.

Reply:
Yea, I wasn't a huge fan of doing it but I didn't want to make the PR more complicated by moving a bunch of existing stuff around.

Reply:
Fair enough. I'm fine if we merge this PR as-is and then we do a "deduplication" PR.